src/test/daml/Daml/Finance/Instrument/Swap/Test/Fpml.daml

```diff
@@ -445,3 +445,175 @@ runStubRateInterpolation = script do
   swapInstrumentAfterFirstPayment <- lifecycleAndVerifySwapPaymentEffects [publicParty] (addDays firstRegularPeriodDate 1) swapInstrument settlers issuer investor observers custodian [observableCid, observableOneMonthCid] expectedConsumedQuantities expectedProducedQuantities
 
   pure ()
+
+-- Test sample trade with two stub periods (intial and final) in the same trade.
+runDualStubSampleTrade : Script ()
+runDualStubSampleTrade = script do
+  [custodian, issuer, investor, calendarDataProvider, publicParty] <-
+    createParties ["Custodian", "Issuer", "Investor", "Calendar Data Provider", "PublicParty"]
+
+  -- Account and holding factory
+  let fp = [("FactoryProvider", singleton publicParty)]
+
+  -- Distribute commercial-bank cash
+  now <- getTime
+  let observers = [("PublicParty", singleton publicParty)]
+  cashInstrumentCid <- Instrument.originate custodian issuer "USD" "US Dollars" observers now
+
+  -- Create and distribute swap
+  -- Fix vs floating interest rate swap: Libor 3M vs 2.0% p.a.
+  let
+    issueDate = date 2022 Sep 14
+    firstRegularPeriodDate = date 2022 Sep 20
+    secondRegularPeriodDate = date 2022 Dec 20
+    thirdRegularPeriodDate = date 2023 Mar 20
+    lastRegularPeriodDate = date 2023 Jun 20
+    maturityDate = date 2023 Sep 14
+    firstRegularPeriodDateFixLeg = date 2022 Sep 20
+    lastRegularPeriodDateFixLeg = firstRegularPeriodDateFixLeg
+    referenceRateId = "USD/LIBOR/3M"
+    referenceRateOneMonthId = "USD/LIBOR/1M"
+    issuerPaysFix = False
+    fixRate = 0.02
+    paymentPeriod = M
+    paymentPeriodMultiplier = 3
+    dayCountConvention = Act360
+    businessDayConvention = ModifiedFollowing
+    observations = M.fromList [(dateToDateClockTime (date 2022 Sep 16), 0.01), (dateToDateClockTime (date 2022 Dec 16), 0.01), (dateToDateClockTime (date 2023 Mar 16), 0.01), (dateToDateClockTime (date 2023 Jun 16), 0.01) ]
+    observationsOneMonth = M.fromList [(dateToDateClockTime (date 2023 Jun 16), 0.02)]
```

Comment:
Yes, this trade uses an extra rate Libor1M for stub rate interpolation (in addition to the main Libor3M rate, which is used for regular periods).

Reply:
aah, understood. 

Can we call the first one `observationsLibor3M` and the second one `observationsLibor1M`? I got confused and thought the OneMonth was referring to the observation date rather than the tenor of the rate.

Reply:
I now fixed this and I also removed a redundant test case (covered by the others, including the new sample trade).